airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -116,7 +116,22 @@ class Orders(IncrementalShopifyStream):
     data_field = "orders"
 
     def path(self, **kwargs) -> str:
-        return f"{self.data_field}.json?status=any"
+        return f"{self.data_field}.json"
+
+    def request_params(self, stream_state=None, next_page_token: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        stream_state = stream_state or {}
+        if next_page_token:
+            params = {"limit": self.limit, **next_page_token}
+        else:
+            params = {
+                "limit": self.limit,
+                "order": f"{self.primary_key} asc",
+                "created_at": self.start_date,
+                "status": "any",
+                # Add state parameter "since_id" for incremental refresh
+                "since_id": stream_state.get(self.cursor_field),
```

Comment:
why are we using `since_id` if we are using `created_at` and pagination?

Reply:
The reason is simple, when we fetch a bunch of data from a particular date, let's say, using updated_at, we might have the situation when a bunch of records are within the same date value but have different ids. Potentially, we might lose the records if we stop for some reason and start over again, we would have the id from where to go on. 
The second reason is to use since_id as the starting point for incremental refresh, after the pagination gets the `next_page_token` we are not using `since_id` anymore, but continue with the defined order of responses from the starting point. I would keep it anyway rather than use some date value in the incremental.

Reply:
Decided to switch to `updated_at` from the `since_id` as it would give us more flexability for future updates and fetching fresh data, if any particular record would be updated, we will catch it to the stream.